mmv1/templates/terraform/examples/big_query_routine_pyspark_mainfile.tf.erb

```diff
@@ -0,0 +1,25 @@
+resource "google_bigquery_dataset" "test" {
+  dataset_id = "<%= ctx[:vars]['dataset_id'] %>"
+}
+
+resource "google_bigquery_connection" "test" {
+  connection_id = "<%= ctx[:vars]['connection_id'] %>"
+  location      = "US"
+  spark { }
+}
+
+resource "google_bigquery_routine" "<%= ctx[:primary_resource_id] %>" {
+  dataset_id      = google_bigquery_dataset.test.dataset_id
+  routine_id      = "<%= ctx[:vars]['routine_id'] %>"
+  routine_type    = "PROCEDURE"
+  language        = "PYTHON"
+  definition_body = ""
```

Comment:
One consideration is how to let the user set `definition_body` when the stored procedure is actually in a separate file(ex. `main_file_uri`). `definition_body` is required as an API, but I think it would be more natural to let it as a non-required property with a blank string as the default.

Reply:
Interesting, I would consider making that change in a separate PR, as it would also affect other use cases for this resource. Adding defaults is relatively easy, but it is a breaking change for us to change or remove them later, so we'd want to be careful here.

IMO, it seems like there could be value to keep it required so that users know they typically need to specify it, and the empty string is a special case to indicate there stored procedure is in a separate file. This also has the benefit of matching the API surface. But you probably know the use cases better, and using a default here could be more ergonomic.